src/lib/draw/GeometryMode.svelte

```diff
@@ -170,16 +164,16 @@
       });
       mapHover.set((results[0]?.id as number) || null);
     }
-  }
+  };
 
-  function onMouseOut() {
-    if (mode == thisMode && currentlyEditing == null) {
+  const onMouseOut = () => {
+    if (currentlyEditing == null) {
       mapHover.set(null);
     }
-  }
+  };
 
-  function onClick(e: MapMouseEvent) {
-    if (mode == thisMode && currentlyEditing == null) {
+  const onClick = (e: MapMouseEvent) => {
+    if (currentlyEditing == null) {
```

Comment:
GeometryMode is the special one. If we're currently editing something, we need to forward events to the right place. So here for example, we need something like

```
if (currentlyEditing == null) {
  ...
} else {
  if (currentlyEditingControls == "point") {
    pointTool.onClick(e);
  } else if (currentlyEditingControls == "snap-polygon" || "route") {
    routeTool.onClick(e);
  }
  ...
}
```

and it'd be most bugproof and least boilerplatey to forward _all_ events to the appropriate place. Not sure yet how to do that nicely though.